rad/src/operators/string.rs

```diff
@@ -35,6 +38,117 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
     item.try_into()
 }
 
+fn magic_map_cleaning(input: BTreeMap<String, RadonTypes>) -> RadonTypes {
+    let map_length = input.len();
+
+    match (map_length, input.get("_text")) {
+        (0, _) => {
+            // <tag></tag>
+            RadonString::from("".to_string()).into()
+        }
+        (1, Some(RadonTypes::Array(rad_array))) => {
+            if rad_array.value().len() == 1 {
+                // <tag>Hello</tag>
+                rad_array.value().into_iter().next().unwrap()
+            } else {
+                // <tag>Hello\nWorld</tag>
+                rad_array.clone().into()
+            }
+        }
+        _ => {
+            let mut new_map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+            for k in input.keys() {
+                match input.get(k) {
+                    Some(RadonTypes::Array(rad_array)) if rad_array.value().len() == 1 => {
+                        let value = rad_array.value().into_iter().next().unwrap();
+                        new_map.insert(k.clone(), value);
+                    }
+                    Some(value) => {
+                        new_map.insert(k.clone(), value.clone());
+                    }
+                    None => unreachable!("Invalid map"),
+                }
+            }
+
+            RadonMap::from(new_map).into()
+        }
+    }
+}
+
+pub fn parse_element_map(
+    input: &minidom::Element,
+    use_namespace: bool,
+    depth: u8,
+) -> Result<RadonTypes, RadError> {
+    if depth > MAX_DEPTH {
+        return Err(RadError::XmlParseOverflow);
+    }
+
+    let mut map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+    if use_namespace {
+        map.insert("_xmlns".to_string(), RadonString::from(input.ns()).into());
+    }
+    for (k, v) in input.attrs() {
+        map.insert(format!("@{}", k), RadonString::from(v).into());
+    }
+
+    for child in input.nodes() {
+        let (ignore, key, value) = match child {
```

Comment:
Instead of `(bool, K, V)` you can use `Option<(K, V)>` to better represent this condition.